app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/MainActivity.kt

```diff
@@ -134,6 +136,13 @@ class MainActivity : BaseActivity(),
         initUserEngagement()
         observeViewModel()
         onSessionAlarmNotificationTapped(intent)
+        showNotificationDisclaimer()
+    }
+
+    private fun showNotificationDisclaimer(){
+        if(AppRepository.readAlarms().isNotEmpty() && !NotificationHelper(applicationContext).notificationsEnabled) {
```

Comment:
Please also "pipe" this UI action through the view model and encapsulate the logic there. This will enable you to write unit test(s) for it.

Reply:
Done in next push, see my comment above about the Unittests :+1:

Reply:
The `AppRepository` is already available as a field in `MainViewModel`.

Reply:
Implemented :+1: 

(Sorry for the double Push, I forgot to commit that line :facepalm:)

Reply:
Don't invoke UI actions (`Toast`) in the view model. This won't be easy to test - you will notice when you write the tests.

Reply:
You forgot to inject the `NotificationHelper` in the `MainViewModel`.

Reply:
I will have a look again once you added the tests. (Please try to push a bit less often)

Reply:
Shown by example in https://github.com/isiko/EventFahrplan/pull/1.